packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -23,12 +25,22 @@ export const accordionHeaderShorthandProps: Array<keyof AccordionHeaderSlots> =
  */
 export const useAccordionHeader = (props: AccordionHeaderProps, ref: React.Ref<HTMLElement>): AccordionHeaderState => {
   const { icon, button, children, expandIcon, inline = false, size = 'medium', expandIconPosition = 'start' } = props;
-  const { onHeaderClick: onAccordionHeaderClick, disabled, open } = useAccordionItemContext();
+  const { onHeaderClick: onAccordionHeaderClick, disabled, open, value } = useAccordionItemContext();
+
+  /**
+   * force disabled state on button if accordion isn't collapsible
+   * and this is the only item opened
+   */
+  const forceDisabled = useContextSelector(
+    AccordionContext,
+    ctx => !ctx.collapsible && ctx.openItems.length === 1 && ctx.openItems[0] === value,
```

Comment:
I wonder why the check for the value of the open item is necessary from the accordion context. If the accordion isn't collapsible, you could just check the value of `open`